client/proxy.go

```diff
@@ -0,0 +1,1066 @@
+
+	proxyConfig.AddConnector(connector)
+
+	mapData, err := proxyConfig.AsConfigMapData()
+	labels := map[string]string{
+		"skupper.io/type": "proxy-definition",
+	}
+	_, err = kube.NewConfigMap(proxyPrefix+proxyName, &mapData, &labels, owner, cli.GetNamespace(), cli.KubeClient)
+	if err != nil {
+		return "", fmt.Errorf("Failed to create proxy config map: %w", err)
+	}
+
+	if options.StartProxy {
+		err = cli.proxyStart(ctx, proxyName)
+		if err != nil {
+			return proxyName, err
+		}
+	}
+	return proxyName, nil
+}
+
+func (cli *VanClient) ProxyDownload(ctx context.Context, proxyName string, downloadPath string) error {
+	certs := []string{"tls.crt", "tls.key", "ca.crt"}
+
+	tarFile, err := os.Create(downloadPath + "/" + proxyName + ".tar.gz")
+	if err != nil {
+		return fmt.Errorf("Unable to create download file: %w", err)
+	}
+
+	// compress tar
+	gz := gzip.NewWriter(tarFile)
+	defer gz.Close()
+	tw := tar.NewWriter(gz)
+	defer tw.Close()
+
+	secret, err := cli.KubeClient.CoreV1().Secrets(cli.GetNamespace()).Get(proxyPrefix+proxyName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to retreive external proxy secret: %w", err)
+	}
+
+	for _, cert := range certs {
+		err = writeTar("qpid-dispatch-certs/conn1-profile/"+cert, secret.Data[cert], time.Now(), tw)
+		if err != nil {
+			return err
+		}
+	}
+
+	configmap, err := kube.GetConfigMap(proxyPrefix+proxyName, cli.GetNamespace(), cli.KubeClient)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve proxy configmap: %w", err)
+	}
+	proxyConfig, err := qdr.GetRouterConfigFromConfigMap(configmap)
+
+	mc, _ := qdr.MarshalRouterConfig(*proxyConfig)
+
+	instance := ProxyInstance{
+		DataDir:  "${QDR_CONF_DIR}",
+		RouterID: "${ROUTER_ID}",
+		Hostname: "${HOSTNAME}",
+	}
+	var buf bytes.Buffer
+	qdrConfig := template.Must(template.New("qdrConfig").Parse(mc))
+	qdrConfig.Execute(&buf, instance)
+
+	if err != nil {
+		return fmt.Errorf("Failed to parse proxy configmap: %w", err)
+	}
+
+	err = writeTar("config/qdrouterd.json", buf.Bytes(), time.Now(), tw)
+	if err != nil {
+		return err
+	}
+
+	proxyInfo := UnitInfo{
+		IsSystemService: false,
+		Binary:          "${QDR_BIN_PATH}",
+		ConfigPath:      "${QDR_CONF_DIR}",
+		ProxyName:       proxyName,
+	}
+
+	qdrUserUnit := serviceForQdr(proxyInfo)
+	err = writeTar("service/"+proxyName+".service", []byte(qdrUserUnit), time.Now(), tw)
+	if err != nil {
+		return err
+	}
+
+	launch := launchScript(proxyInfo)
+	err = writeTar("launch.sh", []byte(launch), time.Now(), tw)
+	if err != nil {
+		return err
+	}
+
+	remove := removeScript(proxyInfo)
+	err = writeTar("remove.sh", []byte(remove), time.Now(), tw)
+	if err != nil {
+		return err
+	}
+
+	expand := expandVars()
+	err = writeTar("expandvars.py", []byte(expand), time.Now(), tw)
+	return nil
+}
+
+func (cli *VanClient) proxyStart(ctx context.Context, proxyName string) error {
+	proxyDir := getDataHome() + "/skupper/" + proxyName
+	svcDir := getConfigHome() + "/systemd/user"
+
```

Comment:
If `$HOME/.local` or `$HOME/.config` do not exist, the call to `startProxyUserService`
fails with:
```
Error: Failed to create start service: Unable to write user unit file: open /home/sample-user/.config/systemd/user/proxy1.service: no such file or directory
```

Seems to resolve by adding:
```
	if err := os.MkdirAll(proxyDir, 0755); err != nil {
		return err
	}
	if err := os.MkdirAll(svcDir, 0755); err != nil {
		return err
	}
```

Reply:
Done. The local share directories get create in setupLocalDir. Was missing the svcDir creation.